editor/editor_inspector.cpp

```diff
@@ -3114,7 +3114,24 @@ void EditorInspector::update_tree() {
 			// Build the doc hint, to use as tooltip.
 
 			// Get the class name.
-			StringName classname = doc_name == "" ? object->get_class_name() : doc_name;
+			StringName classname = doc_name;
+			if (classname == "") {
+				if (Object::cast_to<Resource>(object) && object->get_script()) {
+					Resource *res = cast_to<Resource>(object);
+					if (res) {
+						Script *src = cast_to<Script>(res->get_script());
+						if (src) {
+							// If it is a resource and has a script, grab its script to get the script class.
+							classname = EditorNode::get_editor_data().script_class_get_name(src->get_path());
+						} else {
+							classname = object->get_class_name();
```

Comment:
Also, `classname = object->get_class_name();` can probably show up just after `if (classname == "")`, right? This way it defaults to that, unless `if (res)` and `if (scr)` are true, and you don't need any of the `else` clauses!

Reply:
Indeed!